### PR TITLE
Fix new C++ mangle syntax producing parent as Nspace.

### PR DESCRIPTION
**Pull request by ghost**
@TurkeyMan 
Let me know if this works for you. Do you have a sample code for any other situation? I think maybe an error message?

Don't merge yet, need to look at some other stuff -- just testing.

### CODE REVIEW

File: test/compilable/cppmangle3.d [hunk 1]
```diff
@@ -1,4 +1,6 @@
+module cppmangle3;
 
+import std.traits : fullyQualifiedName;
```
Comment (thewilsonator) [inline test/compilable/cppmangle3.d]: Using Phobos is not allowed in the test suite.  The test below is sufficient.

Comment (Geod24) [reply test/compilable/cppmangle3.d]: In order to test the name, you could also add a test to `fail_compilation` which checks the error message.

**Comment by TurkeyMan** [comment]
This addressed the cases I know how to reproduce.
I caught a few other error messages that printed the wrong string along the way, but I forget what I did to produce them now. trying to remember... but I reckon this will have addressed them too.

File: test/compilable/cppmangle3.d (hunk 1 above)
Comment (TurkeyMan) [reply test/compilable/cppmangle3.d]: That's a good idea.

Comment (look-at-me) [reply test/compilable/cppmangle3.d]: Yah I couldn't think of one either, the only one that I could think of was removed. The symbol clashing lol.

Comment (TurkeyMan) [reply test/compilable/cppmangle3.d]: There are heaps of error messages that do emit the symbol name. I ran into a bunch in short order... but now i've got nothing! ;)

File: src/dmd/dsymbol.d [hunk 2]
```diff
@@ -382,16 +382,30 @@ extern (C++) class Dsymbol : RootObject
         return parent.pastMixin();
     }
 
+    final inout(Dsymbol) pastMixinAndNspace() inout
```
Comment (WalterBright) [inline src/dmd/dsymbol.d]: No Ddoc comment

**Comment by TurkeyMan** [comment]
I mean, you always hassle me to squash my patches before merging them, and I never PR a line if 8 wip's ;)

**Pull request merged**